src/main/java/com/mraof/minestuck/world/lands/terrain/WoodLandType.java

```diff
@@ -20,67 +22,123 @@
 import net.minecraft.world.level.levelgen.placement.CountPlacement;
 import net.minecraft.world.level.levelgen.placement.HeightRangePlacement;
 import net.minecraft.world.level.levelgen.placement.InSquarePlacement;
+import net.minecraft.world.level.levelgen.structure.Structure;
+import net.minecraft.world.level.levelgen.structure.StructureSet;
+import net.minecraft.world.level.levelgen.structure.placement.RandomSpreadStructurePlacement;
+import net.minecraft.world.level.levelgen.structure.placement.RandomSpreadType;
+
+import java.util.function.Consumer;
 
 public class WoodLandType extends TerrainLandType
 {
 	public static final String WOOD = "minestuck.wood";
-	public static final String OAK = "minestuck.oak";
-	public static final String LUMBER = "minestuck.lumber";
+	public static final String PLANKS = "minestuck.planks";
+	public static final String CARVINGS = "minestuck.carvings";
 	
 	public WoodLandType()
 	{
-		super(new Builder(MSEntityTypes.SALAMANDER).names(WOOD, OAK, LUMBER)
+		super(new Builder(MSEntityTypes.SALAMANDER).names(WOOD, PLANKS, CARVINGS)
 				.skylight(1/2F).fogColor(0.0, 0.16, 0.38).skyColor(0.0, 0.3, 0.4)
 				.music(MSSoundEvents.MUSIC_WOOD));
 	}
 	
 	@Override
 	public void registerBlocks(StructureBlockRegistry registry)
 	{
-		registry.setBlock("upper", Blocks.OAK_LOG);
-		registry.setBlock("surface", MSBlocks.TREATED_PLANKS);
-		registry.setBlock("structure_primary", Blocks.JUNGLE_WOOD);
-		registry.setBlock("structure_primary_decorative", Blocks.DARK_OAK_LOG);
-		registry.setBlock("structure_primary_stairs", Blocks.DARK_OAK_STAIRS);
-		registry.setBlock("structure_secondary", Blocks.JUNGLE_PLANKS);
-		registry.setBlock("structure_secondary_decorative", Blocks.DARK_OAK_PLANKS);
-		registry.setBlock("structure_secondary_stairs", Blocks.JUNGLE_STAIRS);
-		registry.setBlock("light_block", MSBlocks.GLOWING_WOOD);
-		registry.setBlock("bush", Blocks.RED_MUSHROOM);
+		registry.setBlock("ground", MSBlocks.UNCARVED_WOOD);
+		registry.setBlock("upper", MSBlocks.CHIPBOARD);
```

Comment:
I dont think it makes sense to have chipboard for upper since its sandwhiched between uncarved wood. Its okay to also make it uncarved

Reply:
Fixed